maven-release-manager/src/main/java/org/apache/maven/shared/release/phase/AbstractRewritePomsPhase.java

```diff
@@ -517,128 +517,177 @@ private void rewriteArtifactVersions(
+            String property,
+            Properties properties,
+            ReleaseResult result,
+            ReleaseDescriptor releaseDescriptor)
+            throws ReleaseFailureException {
+        if (properties == null) {
+            logInfo(
+                    result,
+                    "  Ignoring artifact version update for " + artifactKey + " as expression " + rawVersion
+                            + " cannot be locally resolved");
+            return false;
+        }
+        // check for value of property
+        boolean isUpdated = false;
+        String propertyValue = properties.getProperty(property);
+        if (propertyValue != null) {
+            if (propertyValue.equals(originalVersion)) {
+                logInfo(result, "  Updating " + rawVersion + " to " + mappedVersion);
+                // change the property only if the property is the same as what's in the reactor
+                properties.setProperty(property, mappedVersion);
+                isUpdated = true;
+            } else if (mappedVersion.equals(propertyValue)) {
+                // this property may have been updated during processing a sibling.
+                logInfo(
+                        result,
+                        "  Ignoring artifact version update for expression " + rawVersion
+                                + " because it is already updated");
+            } else if (!mappedVersion.equals(rawVersion)) {
+                // WARNING: ${pom.*} prefix support and ${version} is about to be dropped in mvn4!
+                // https://issues.apache.org/jira/browse/MNG-7404
+                // https://issues.apache.org/jira/browse/MNG-7244
+                if (mappedVersion.matches("\\$\\{project.+\\}")
+                        || mappedVersion.matches("\\$\\{pom.+\\}")
+                        || "${version}".equals(mappedVersion)) {
+                    logInfo(result, "  Ignoring artifact version update for expression " + mappedVersion);
+                    // ignore... we cannot update this expression
+                } else {
+                    // the value of the expression conflicts with what the user wanted to release
+                    throw new ReleaseFailureException("The artifact (" + artifactKey + ") requires a "
+                            + "different version (" + mappedVersion + ") than what is found ("
+                            + propertyValue + ") for the expression (" + rawVersion + ") in the "
+                            + "project (" + projectKey + ").");
+                }
+            }
+        } else {
+            if (CI_FRIENDLY_PROPERTIES.contains(property)) {
+                logInfo(result, "  Ignoring artifact version update for CI friendly expression " + rawVersion);
             } else {
-                // artifact not related to current release
+                // the expression used to define the version of this artifact may be inherited
+                logInfo(
```

Comment:
maybe logWarn is actually better here.

Reply:
We need to take into account the fact that modelVersion 4.1.0 will allow users to only define the version in the root parent, all other versions will be inferred.  I would not want tons of warnings when using such setup.
If this is a supported use case, I don't see why a warning would be legit.